cdap-app-fabric/src/main/java/co/cask/cdap/app/guice/AuthorizationModule.java

```diff
@@ -128,16 +129,19 @@ public DynamicDatasetCache get() {
   private static final class AdminProvider implements Provider<Admin> {
     private final DatasetFramework dsFramework;
     private final SecureStoreManager secureStoreManager;
+    private final NamespaceAdmin namespaceAdmin;
 
     @Inject
-    private AdminProvider(DatasetFramework dsFramework, SecureStoreManager secureStoreManager) {
+    private AdminProvider(DatasetFramework dsFramework, SecureStoreManager secureStoreManager,
+                          NamespaceAdmin namespaceAdmin) {
```

Comment:
Do you need the `NamespaceAdmin`? Can you use `NamespaceQueryAdmin` instead? I remember there was some complicate circular guice module issue when `NamespaceAdmin` is being used.

Reply:
Nope don't need the full NamespaceAdmin, didn't know NamespaceQueryAdmin existed. Will change.